driver.go

```diff
@@ -542,6 +546,16 @@ const (
 	PartitionedNonAtomic
 )
 
+func (c *conn) UnderlyingClient() (*spanner.Client, error) {
+	if c.inTransaction() {
+		return nil, spanner.ToSpannerError(status.Error(codes.FailedPrecondition, "cannot access *spanner.Client when in a transaction"))
+	}
+	if c.batch != nil {
+		return nil, spanner.ToSpannerError(status.Error(codes.FailedPrecondition, "cannot access *spanner.Client with an active batch"))
+	}
```

Comment:
Purely from a technical point of view, I don't think we need this restriction, although I understand it from a conceptual viewpoint; the client that is returned cannot be used to interact with the current transaction on the connection. There is however nothing wrong with using the client that is returned to execute other transactions or queries.

Reply:
Oh sure, I would be completely fine removing the limitation.

Reply:
Any opinion whether it should be:

```
UnderlyingClient() (client *spanner.Client, err error)

or

UnderlyingClient() (client any, err error)
```

Reply:
I would say `UnderlyingClient() (client *spanner.Client, err error)`. Anyone calling this method does so with the intention of using Spanner-specific features, so declaring that the function returns a Spanner-specific type seems reasonable to me.